src/dodal/devices/oav/pin_image_recognition/__init__.py

```diff
@@ -15,20 +13,32 @@
     ScanDirections,
     identity,
 )
-from dodal.devices.ophyd_async_utils import create_soft_signal_rw
+from dodal.devices.ophyd_async_utils import create_soft_signal_r, create_soft_signal_rw
 from dodal.log import LOGGER
 
+Tip = tuple[Optional[int], Optional[int]]
+
+
+def set_pin_value(*args, **kwargs):
+    if args[1] == PinTipDetection.INVALID_POSITION:
+        raise
```

Comment:
Don't you need to create some Exception here to raise? Even if this works somehow, I think you should raise a custom exception here so that the warning message later is more informative whether it was an invalid position or some other problem

Reply:
`raise` just raises a generic `Exception`, and the try block catches a generic exception. But yes, I agree, might as well make a custom exception for this